### ISSUE TITLE
⛔ 🚀 Background image size is missing when using a dynamic tag

### ISSUE THREAD

**Issue by perks98** [elementor/elementor#21359]
### Prerequisites

- [X] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [X] The issue still exists against the latest stable version of Elementor.

### Description

When using a dynamic tag to pull in an image like the featured image there are no image size options, while without using any tags you get the option to adjust the size.

![Screenshot 2023-02-15 202136](https://user-images.githubusercontent.com/55829113/219145083-f565c7c6-3b09-42c8-b283-6cae7bcd3e09.png)

![Screenshot 2023-02-15 202231](https://user-images.githubusercontent.com/55829113/219145117-26b12580-8c91-4f48-8b63-7ef0f69ee1e7.png)

Also when you add an image to the background first to make the size options appear then decide to use a dynamic tag for an image the size option says there but is not used in final frontend image render.

See steps to reproduce section for more info.

### Steps to reproduce

**Using a dynamic tag right away doesn't present the image size option:**

https://user-images.githubusercontent.com/55829113/219142110-cc0c5608-163e-434b-9dbb-f172fa8eadde.mp4

 
<hr>
 
**Choosing an image first then dynamic tag doesn't respect the size chosen in preview or frontend (should be thumbnail size in example):**

https://user-images.githubusercontent.com/55829113/219142338-455bfc54-bed7-479d-8205-d55e91e190a9.mp4

### Isolating the problem

- [X] This bug happens with only Elementor plugin active (and Elementor Pro).
- [X] This bug happens with a Blank WordPress theme active ([Hello theme](https://wordpress.org/themes/hello-elementor/)).
- [X] I can reproduce this bug consistently following the steps above.

### System Info

<details>
<summary>System Info</summary>

````txt

					
== Server Environment ==
	Operating System: Linux
	Software: Apache
	MySQL version: MariaDB Server v10.4.26
	PHP Version: 7.4.33
	PHP Memory Limit: 128M
	PHP Max Input Vars: 2500
	PHP Max Post Size: 200M
	GD Installed: Yes
	ZIP Installed: Yes
	Write Permissions: All right
	Elementor Library: Connected

== WordPress Environment ==
	Version: 6.1.1
	Site URL: [private]
	Home URL: [private]
	WP Multisite: No
	Max Upload Size: 200 MB
	Memory limit: 40M
	Max Memory limit: 512M
	Permalink Structure: /%postname%/
	Language: en-GB
	Timezone: Europe/London
	Admin Email: [private]
	Debug Mode: Inactive

== Theme ==
	Name: Astra Child
	Version: 1.0.0
	Author: Brainstorm Force
	Child Theme: Yes
	Parent Theme Name: Astra
	Parent Theme Version: 4.0.2
	Parent Theme Author: Brainstorm Force

== User ==
	Role: administrator
	WP Profile lang: en_GB
	User Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/110.0.0.0 Safari/537.36

== Active Plugins ==
	Astra Pro
		Version: 4.0.1
		Author: Brainstorm Force

	Better Search Replace
		Version: 1.4.2
		Author: WP Engine

	CookieYes | GDPR Cookie Consent
		Version: 3.0.8
		Author: CookieYes

	Elementor
		Version: 3.11.1
		Author: Elementor.com

	Elementor Pro
		Version: 3.11.1
		Author: Elementor.com

	FluentSMTP
		Version: 2.2.4
		Author: FluentSMTP & WPManageNinja Team

	Rank Math SEO
		Version: 1.0.108
		Author: Rank Math

	Redirection
		Version: 5.3.9
		Author: John Godley

	WooCommerce
		Version: 7.4.0
		Author: Automattic

	WooCommerce Handepay
		Version: 1.2
		Author: Handepay

	WooCommerce PayPal Payments
		Version: 2.0.2
		Author: WooCommerce


== Must-Use Plugins ==
	StackCache
		Version: 
		Author: Stack CP


== Features ==
	Custom Fonts: 0
	Custom Icons: 0

== Integrations ==
	
	recaptcha_v3: Active
	woocommerce: Active


== Elementor Experiments ==
	Optimised DOM Output: Active by default
	Improved asset loading: Active by default
	Improved CSS Loading: Active by default
	Inline Font Icons: Inactive by default
	Accessibility improvements: Active by default
	Additional Custom Breakpoints: Active by default
	admin_menu_rearrangement: Inactive by default
	Flexbox Container: Inactive by default
	Upgrade Swiper Library: Inactive by default
	Default to New Theme Builder: Active by default
	Editor V2: Inactive by default
	Landing pages: Active by default
	Nested Elements: Inactive by default
	Lazy Load Background Images: Inactive by default
	Save as Default: Active by default
	Page Transitions: Active by default
	Notes: Active by default
	Loop: Active by default
	Form Submissions: Active by default
	Scroll Snap: Active by default
	Mega Menu: Inactive by default


== Log ==
	
Log: showing 20 of 372023-01-31 09:45:22 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.10.2',
  'to' => '3.10.3',
)]
2023-02-13 11:34:21 [info] elementor::elementor_updater Started 
2023-02-13 11:34:21 [info] Elementor/Upgrades - _on_each_version Start  
2023-02-13 11:34:21 [info] Elementor/Upgrades - _on_each_version Finished 
2023-02-13 11:34:21 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.10.2',
  'to' => '3.11.0',
)]
2023-02-13 11:34:26 [info] elementor-pro::elementor_pro_updater Started 
2023-02-13 11:34:26 [info] Elementor Pro/Upgrades - _on_each_version Start  
2023-02-13 11:34:26 [info] Elementor Pro/Upgrades - _on_each_version Finished 
2023-02-13 11:34:26 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.10.3',
  'to' => '3.11.0',
)]
2023-02-15 19:53:38 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor',
  'from' => '3.11.0',
  'to' => '3.11.1',
)]
2023-02-15 19:53:39 [info] elementor::elementor_updater Started 
2023-02-15 19:53:39 [info] Elementor/Upgrades - _on_each_version Start  
2023-02-15 19:53:39 [info] Elementor/Upgrades - _on_each_version Finished 
2023-02-15 19:53:39 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.11.0',
  'to' => '3.11.1',
)]
2023-02-15 19:53:41 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor',
  'from' => '3.11.0',
  'to' => '3.11.1',
)]
2023-02-15 19:54:04 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.11.0',
  'to' => '3.11.1',
)]
2023-02-15 19:54:04 [info] elementor-pro::elementor_pro_updater Started 
2023-02-15 19:54:04 [info] Elementor Pro/Upgrades - _on_each_version Start  
2023-02-15 19:54:04 [info] Elementor Pro/Upgrades - _on_each_version Finished 
2023-02-15 19:54:04 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.11.0',
  'to' => '3.11.1',
)]

PHP: showing 3 of 3PHP: 2023-02-06 10:15:49 [notice X 22][/home/sites/21b/b/ba92b594de/public_html/wp-content/plugins/elementor/includes/conditions.php::90] Trying to access array offset on value of type null [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2023-02-13 09:58:52 [notice X 2][/home/sites/21b/b/ba92b594de/public_html/wp-content/plugins/elementor/core/common/modules/ajax/module.php::172] Undefined index: data [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2023-02-13 11:21:46 [notice X 1][/home/sites/21b/b/ba92b594de/public_html/wp-content/plugins/elementor-pro/modules/theme-builder/classes/conditions-manager.php::168] Undefined index: editor_post_id [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]

JS: showing 2 of 2JS: 2023-02-06 10:49:30 [error X 2][[private]/wp-admin/load-scripts.php?c=0&loadchunk_0=underscore,jquery-core,jquery-migrate,backbone,jquery-ui-core,jquery-ui-mouse,wp-api-request,regenerator-runtime,wp-polyfill,wp-&loadchunk_1=hooks&ver=6.1.1:4:51824] Cannot read properties of undefined (reading 'replace') 
JS: 2023-02-13 11:20:51 [error X 1][[private]/wp-content/plugins/elementor/assets/js/editor.min.js?ver=3.11.0:3:1087291] Cannot read properties of null (reading 'getBoundingClientRect') 



== Elementor - Compatibility Tag ==
	
	Elementor Pro: Compatible

== Elementor Pro - Compatibility Tag ==
	
````
				
</details>

**Comment by DavidGMiles** [comment]
Yup crazy ommission, this makes the new feature half baked

**Comment by nicholaszein** [comment]
Hi @perks98! 

Thank you for submitting your issue! 🙏 

Please allow us some time to investigate it, and get back to you with our findings. 🧐 

Best regards

**Comment by ShadowXVII** [comment]
Yep, totally missed the mark on that one. They locked previous thread (#6778) which is a pain.

**Comment by DavidGMiles** [comment]
Total fail so far - half baked feature - no progress - this is everything that is wrong with Elementor

**Comment by DavidGMiles** [comment]
Is this being fixed or just left as is?

**Comment by perks98** [comment]
Not fixed in the 3.12 beta and I don't think there are any pull requests to fix so... left as is for now.

**Comment by DavidGMiles** [comment]
> Not fixed in the 3.12 beta and I don't think there are any pull requests to fix so... left as is for now.

So half develop a feature, release, make a big fanfare about it, have bugs pointed out, and ignore them!

**Comment by ShadowXVII** [comment]
My request to Elementor support:

> Kindly be advised that we are already aware of this and our developers are currently working to fix it.

> You will be notified once it is fixed.

> Currently, we do not have an exact ETA for when this issue will be solved, but once the bug is fixed, we will update the status in our Changelog Page.

**Comment by perks98** [comment]
I don't use Elementor support anymore as most issues I raise their developers are aware of but never have an ETA on fixing ~ like #14743 & #18874 OR the excuse is that it has been an issue for so long that users have found ways around it and fixing it now will break that ~ like #20041. These days I try only to report stuff around beta season as they are more likely to action or if it is a big major issue. Unfortunately 3.11 is already out so I don't think they will action this any time soon unless we have 50 users demand for it. ~~ but I could be wrong

**Comment by DavidGMiles** [comment]
5 years for a solution which is incomplete - it kind of sums Elementor up - if it’s not new and shiny they lose interest and leave bugs to fester

**Comment by caspervoogt** [comment]
Crazy that image sizes are still not possible with dynamic background images. That makes them much less useful. Super frustrating.

**Comment by DavidGMiles** [comment]
Hey stated that it wasn’t important to them and they didn’t see any need for it - which sucks

**Comment by DavidGMiles** [comment]
Will this half-completed feature ever be finished - it is useless as it stands - the ability to select Background Image Size Options when using a Dynamic Image source is really important for performance

**Comment by VerdiH** [comment]
Any news @nicholaszein just got some questions about it

**Comment by DavidGMiles** [comment]
> Any news @nicholaszein just got some questions about it

Evidently not - another one into the black hole I guess

**Comment by nicholaszein** [comment]
@DavidGMiles and @VerdiH please refer to Amitai's section on Elementor's Q3 Roadmap Event. It should clear out any doubt about how we are dealing with bug reports. 

https://www.youtube.com/live/PHqbPU-h0NY?feature=share&t=1316

@DavidGMiles please, if you have any complaints towards the Elementor Company, use the following channel:

https://elementor.com/contact

This is not the place for this kind of comment.

Thank you very much.

**Comment by sixgunnar** [comment]
This feature is definitely needed. Was surpised to find that the image size selection was added for background images in 3.11 because I could not see this option. Mostly use background images loaded from post featured image..

**Comment by DavidGMiles** [comment]
When I recently asked Elementor Support about this I was told 'This is not classified as a bug by our support teams, this is intended functionality'

**Comment by danielvigh** [comment]
+1

**Comment by perks98** [comment]
> Hi @perks98!
> 
> Thank you for submitting your issue! 🙏
> 
> Please allow us some time to investigate it, and get back to you with our findings. 🧐
> 
> Best regards

Hi @nicholaszein,
Have you or your team had a chance yet to 'investigate' this yet? 🤔

**Comment by perks98** [comment]
#24146 <-- not sure if that potential code relates to this issue but if it is then it looks like this issue may not get fixed and just a message of "Image size settings don’t apply to Dynamic Images." instead which isn't really a solution.

**Comment by DavidGMiles** [comment]
Brilliant! What an outstanding solution!

**Comment by TakacsAdrian** [comment]
+1

**Comment by ZoltanGalCMDesign** [comment]
+1

**Comment by Leadsquad** [comment]
OMG, that's so dumb, and they asklng 59/year

**Comment by danielvigh** [comment]
Any news about this @nicholaszein ?

**Comment by zanga7** [comment]
Anything for this?

**Comment by alexplex** [comment]
I'm also awaiting a fix for this.

**Comment by brajno** [comment]
This has a huge impact on performance.

**Comment by ncasedk** [comment]
Damn. Still no fix. This is VERY problematic

**Comment by zanga7** [comment]
Anything? Any work arounds?

**Comment by miguel-cravidao** [comment]
I would also like to have an update on this. Still using full sized images in every dynamic element I do.

I tried as workaround to use images but I can't make images center/fit as well as background images. I define min-height for images and object fit to center center but i can't get images center/cover properly like in background images.

**Comment by v3ctordesign** [comment]
So it's been 624 days and still nothing...

**Comment by caspervoogt** [comment]
This is really a problem. I am using (and loving) loop grids and setting background images using dynamic tags, but can't use an image size with that. I have to use a workaround consisting of absolutely positioned Image elements and a bunch of custom CSS. 

[In this video I show the issue](https://www.loom.com/share/21ef521978ff44a7810fafa7b5bbc648), along with my workaround using Image element and some absolutely positioned elements. This is not an ideal workaround at all; it serves to illustrate why the background image option desperately needs an image size option.

**Comment by miguel-cravidao** [comment]
Yes @caspervoogt tks for sharing but I refuse to go that path to do something so simple, unless i need have a project that requires some high degree of performance. Right now, with image compression and CDN i can live with that.

**Comment by caspervoogt** [comment]
@miguel-cravidao I know, it is ridiculous isn't it. Granted, it didn't take me long to do (because I am so used to having to use this workaround by now) ... but it's silly and sad to have to jump through such hoops.

**Comment by nicholaszein** [comment]
Hey everyone!

An update to this issue:

Initially this functionality was not intended to be implemented, and images loaded via dynamic tags couldn't have the resolution set. We intended to update the controls to inform this limitation by adding a note below the Background Image selector indicating this was not possible:

> _"Dynamic background images aren’t supported by Image Resolution settings"_

However, due to the demand, we added this to our roadmap as a new feature.

Unfortunately, after performing various tests we concluded implementing this feature could have a heavy impact on performance both inside the Editor, as well as on the frontend during the first rendering of the page, when the CSS is being generated. So we placed this feature on hold until we improved our infrastructure.

But we have great news! 🙌 

### Support for this feature will be considered for Elementor v4, after all dynamic capabilities are implemented, in a future phase!

For this reason, we are closing this issue, and merging it with the Feature Request [#22750](https://github.com/orgs/elementor/discussions/22750).

Please join the discussion there, and cast your vote as well!

Thank you all for your feedback! 🙏
